panic: Unhandled case in Node.Text: *ast.ComputedPropertyName

panic: Unhandled case in Node.Text: *ast.ComputedPropertyName

goroutine 270767 [running]:
github.com/microsoft/typescript-go/internal/ast.(*Node).Text(0x142382e0960?)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/ast/ast.go:336 +0x3e4
github.com/typescript-eslint/tsgolint/internal/rules/unbound_method.init.func1.3(0x142382dc3c0, 0x142382e0960, 0x14238295a40, 0x1)
        /home/runner/work/tsgolint/tsgolint/internal/rules/unbound_method/unbound_method.go:242 +0x84
github.com/typescript-eslint/tsgolint/internal/rules/unbound_method.init.func1.5(0x142382dc3c0)
        /home/runner/work/tsgolint/tsgolint/internal/rules/unbound_method/unbound_method.go:281 +0xfc
github.com/typescript-eslint/tsgolint/internal/linter.RunLinterOnProgram.func1.6(0x8a2, 0x142382dc3c0)

This is from an internal codebase, I have access to it, will try to reproduce.

--- a/internal/rules/unbound_method/unbound_method.go
+++ b/internal/rules/unbound_method/unbound_method.go
@@ -237,6 +237,11 @@ var UnboundMethodRule = rule.Rule{
 		}
 
 		checkBindingProperty := func(patternNode *ast.Node, initNode *ast.Node, propertyName *ast.Node, parentIsAssignmentPatternLike bool) {
+			// Skip computed property names as they cannot be statically analyzed
+			if ast.IsComputedPropertyName(propertyName) {
+				return
+			}
+			
 			if initNode != nil {
 				if !isNativelyBound(initNode, propertyName) {
 					reported := checkIfMethodAndReport(propertyName, checker.Checker_getPropertyOfType(ctx.TypeChecker, ctx.TypeChecker.GetTypeAtLocation(initNode), propertyName.Text()))

--- a/test_computed_property.js
+++ b/test_computed_property.js
@@ -1,0 +1,17 @@
+// Test case from issue #226
+const FOO = 'foo';
+const BAR = 'bar';
+
+// This should not crash anymore with computed property names
+({ [FOO]: BAR } = {});
+
+// Additional test cases with computed properties
+const obj = {
+  [FOO]: function() {
+    return this;
+  },
+  [BAR]: () => {},
+};
+
+// Destructuring with computed property
+const result = ({ [FOO]: value } = { foo: 'test' });

